cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityOperator.java

```diff
@@ -348,6 +358,49 @@ public ServiceAccount generateServiceAccount() {
         return super.generateServiceAccount();
     }
 
+    @Override
+    protected String getRoleName() {
+        return getRoleName(cluster);
+    }
+
+    /**
+     * Get the name of the Entity Operator Role given the name of the {@code cluster}.
+     * @param cluster The cluster name
+     * @return The name of the EO role.
+     */
+    public static String getRoleName(String cluster) {
+        return entityOperatorName(cluster);
+    }
+
+    /**
+     * Read the entity operator ClusterRole, and use the rules to create a new Role.
+     * This is done to avoid duplication of the rules set defined in source code.
+     *
+     * @param namespace the namespace this role will be located
+     *
+     * @return role for the entity operator
+     */
+    public Role generateRole(String namespace) {
+        List<PolicyRule> rules;
+
+        try (BufferedReader br = new BufferedReader(
+                new InputStreamReader(
+                    Main.class.getResourceAsStream("/cluster-roles/031-ClusterRole-strimzi-entity-operator.yaml"),
+                    StandardCharsets.UTF_8)
+            )
+        ) {
+            String yaml = br.lines().collect(Collectors.joining(System.lineSeparator()));
+            ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+            ClusterRole cr = yamlReader.readValue(yaml, ClusterRole.class);
```

Comment:
Any reason why we're not just using a `YAMLMapper` here? That would probably avoid the need for `yaml`, since it can take an `InputStream` to parse.

Reply:
Oh, well I copied this bit from a similar bit of code in the System tests - not really messed with YAML too much in Java, so didn't know about the alternative!